builtin/joined_dataset.cc

```diff
@@ -181,115 +159,8 @@ struct JoinedDataset::Itl
             // We can use a fast path, since we have simple non-filtered
             // equijoin
 
-            // Where expressions for the left and right side
-            auto runSide = [&] (const AnnotatedJoinCondition::Side & side,
-                                const Dataset & dataset)
-                -> std::vector<std::tuple<ExpressionValue, RowName, RowHash> >
-                {
-                    auto generator = dataset.queryBasic
-                    (context, side.select, side.when, *side.where, side.orderBy,
-                     0, -1, true /* allowParallel */);
-                    auto rows = generator(-1);
-                
-                    if (debug)
-                        cerr << "got rows " << jsonEncode(rows) << endl;
-                    
-                    // Now we extract all values and re-sort.  This is necessary because
```

Comment:
Could I get a comment on that @jeremybarnes ? Now that I put it in an embedding I explicitely get a single value not a row. Im not sure how multiple values could have happened before and it doesnt break any test.